docs/integrations/webhooks/create-a-webhook.mdx

```diff
@@ -0,0 +1,361 @@
+  if (req.method !== 'POST') {
+    return res.status(405)
+  }
+  // You can find this in the Clerk Dashboard -> Webhooks -> choose the webhook
+  const WEBHOOK_SECRET = process.env.WEBHOOK_SECRET
+
+  if (!WEBHOOK_SECRET) {
+    throw new Error('Please add WEBHOOK_SECRET from Clerk Dashboard to .env or .env.local')
+  }
+
+  // Get the headers
+  const svix_id = req.headers["svix-id"] as string;
+  const svix_timestamp = req.headers["svix-timestamp"] as string;
+  const svix_signature = req.headers["svix-signature"] as string;
+
+
+  // If there are no headers, error out
+  if (!svix_id || !svix_timestamp || !svix_signature) {
+    return res.status(400).json({ error: 'Error occured -- no svix headers' })
+  }
+
+  console.log('headers', req.headers, svix_id, svix_signature, svix_timestamp)
+  // Get the body
+  const body = (await buffer(req)).toString()
+
+  // Create a new Svix instance with your secret.
+  const wh = new Webhook(WEBHOOK_SECRET);
+
+  let evt: WebhookEvent
+
+  // Verify the payload with the headers
+  try {
+    evt = wh.verify(body, {
+      "svix-id": svix_id,
+      "svix-timestamp": svix_timestamp,
+      "svix-signature": svix_signature,
+    }) as WebhookEvent
+  } catch (err) {
+    console.error('Error verifying webhook:', err);
+    return res.status(400).json({ 'Error': err })
+  }
+
+  // Do something with the payload
+  // For this guide, you simply log the payload to the console
+  const { id } = evt.data;
+  const eventType = evt.type;
+  console.log(`Webhook with and ID of ${id} and type of ${eventType}`)
+  console.log('Webhook body:', body)
+
+  return res.status(200).json({ response: 'Success' })
+}
+```
+</CodeBlockTabs>
+
+## Add your endpoint to Middleware
+
+Your Route Handler must be made public or ignored by Middleware to allow the request to succeed - otherwise it will redirect to sign in like any other protected route. The following example will make any webhooks created under `/api/webhooks/` public. See [authMiddleware](/docs/references/nextjs/auth-middleware) for more information.
+
+```tsx filename="middleware.tsx" {4}
+import { authMiddleware } from "@clerk/nextjs";
+
+export default authMiddleware({
+  publicRoutes: ["/api/webhooks(.*)"],
+});
+
+export const config = {
+  matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
+};
+```
+
+## Test your webhook
+
+1. Start your Next.js server.
+2. On your endpoint's settings page in the Clerk Dashboard, select the **Testing** tab.
+3. In the **Select event** dropdown, select `user.updated`.
+4. Select the **Send Example** button.
+5. Below that section, in the **Message Attempts** section, you should see a successful attempt with a status of `200`.
+
+### Message failed
+
+If the message failed:
+
+1. Select the message.
+2. Scroll down to the **Webhook Attempts** section.
+3. Select the arrow icon the left side.
+4. Investigate the error. Your solution is going to depend on the error message.
+
+## Add logic for updating a user
```

Comment:
I think this section is unneeded. Updating a user in the Dashboard, or via the `<UserProfile />` will trigger `user.updated` and reduce complexity. I would suggest dropping this section in lieu of the next suggestion.

If this remains in the guide, then one of the two following things should happen:

- the code example should include a form to change the name
- the triggering section should mention that the code needs to be edited to change the name for subsequent tests